common/decklist.cpp

```diff
@@ -463,32 +465,84 @@ bool DeckList::saveToFile_Native(QIODevice *device)
     return true;
 }
 
+void DeckList::removeEmptyLinesFromStart(QVector<QString> &inputs)
+{
+    // https://stackoverflow.com/questions/4642339/trimming-a-vector-of-strings
+    inputs.erase(inputs.begin(), std::find_if(inputs.begin(), inputs.end(), [](const QString& s) {return !s.isEmpty();}));
+}
+
 bool DeckList::loadFromStream_Plain(QTextStream &in)
 {
     cleanList();
+    bool inSideboard = false, titleFound = false, isSideboard;
+    int okRows = 0, blankLines = 0;
+    QVector<QString> inputs;
+
+    // Lets convert input QTextStream => QVector
+    while (!in.atEnd())
+        inputs.push_back(in.readLine().simplified().toLower());
+
+    // "Trim" the inputs of all blank lines at the start & end of paste
+    removeEmptyLinesFromStart(inputs);
+    std::reverse(inputs.begin(), inputs.end()); // Reverse vector (to remove blanks from end)
+    removeEmptyLinesFromStart(inputs);
+    std::reverse(inputs.begin(), inputs.end()); // Vector now in correct order
+
+    // This will "trim" multiple blank lines in a row to just one blank
```

Comment:
I usually see the english term "condense" for such operations, maybe it fits better?